find_ndjson.go

```diff
@@ -280,22 +285,28 @@ func (s *server) doFindNDJson(ctx context.Context, w http.ResponseWriter, source
 	}()
 
 	var rs resultStats
+	var foundCaskade, foundRegular bool
 LOOP:
 	for {
 		select {
 		case <-ctx.Done():
 			break LOOP
-		case result, ok := <-resultsChan:
+		case rwb, ok := <-resultsChan:
 			if !ok {
 				break LOOP
 			}
+			result := rwb.rslt
 			absent := results.putIfAbsent(result)
 			if !absent {
 				continue
 			}
 
 			rs.observeResult(result)
 
+			_, isCaskade := rwb.bknd.(caskadeBackend)
+			foundCaskade = foundCaskade || isCaskade
+			foundRegular = foundRegular || !isCaskade
```

Comment:
are we separating cascade dht. and cascade legacy-bitswap?

Reply:
Not for this metric. The main purpose of it is to calculate how many requests can be satisfied by `storetheindex` (without cascading).

Reply:
it would be great to use the same setup to know how many requests are satisfied by the DHT as a specific source as well

Reply:
We capture those already from `cassette` and from `caskadht`. What we've been missing - is the same metric for `storetheindex`. I've added it to `storetheindex` first but then realised that all records exist in at least 2 indexers. We can't simply capture a `sum` of all 200s from `storetheindex` and  treat it as a success rate of the indexers (it'd be over reporting).